c/src/core/engine.c

```diff
@@ -2299,7 +2314,11 @@ int pn_condition_copy(pn_condition_t *dest, pn_condition_t *src) {
   assert(src);
   int err = 0;
   if (src != dest) {
+    if (src->name == NULL) src->name = pn_string(NULL);
```

Comment:
FWIW `pn_string_copy()` will cope correctly with NULL `src->name` string (but not NULL `dest->name`).
Perhaps if src is NULL and if dest is not NULL the `dest->name` should be freed?